Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -186,6 +187,49 @@ internal set
         // keep track of all sceneIds to detect scene duplicates
         static readonly Dictionary<ulong, NetworkIdentity> sceneIds = new Dictionary<ulong, NetworkIdentity>();
 
+
+
+        /// <summary>
+        /// This is invoked on clients when the server has caused this object to be destroyed.
+        /// <para>This can be used as a hook to invoke effects or do client specific cleanup.</para>
+        /// </summary>
+        ///<summary>Called on clients when the server destroys the GameObject.</summary>
+        public UnityEvent OnNetworkDestroy = new UnityEvent();
```

Comment:
Let's reorder the events in this class the same order they are invoked.  This will make it more intuitive to know which one to hook up to.

Reply:
I copied the order from NetworkBehaviour :P. Moved OnNetworkDestroy to the bottom. see below.